plugins/remote-config/frontend/public/javascripts/countly.views.js

```diff
@@ -13,51 +18,247 @@
+                var results = [];
+                data.forEach(function(value) {
+                    var ob = {
+                        value: value
+                    };
+                    results.push(ob);
+                });
+                return cb(results);
+            },
+            createFilter: function(queryString) {
+                return function(value) {
+                    return value.toLowerCase().indexOf(queryString.toLowerCase()) === 0;
+                };
+            },
             onSubmit: function(doc) {
                 var self = this;
                 var action = "countlyRemoteConfig/parameters/create";
                 if (doc._id) {
                     action = "countlyRemoteConfig/parameters/update";
                 }
-
+                if (!doc.description || !this.showDescription) {
+                    doc.description = "-";
+                }
+                if (doc.status === "Expired" && doc.expiry_dttm > Date.now()) {
+                    doc.status = "Running";
+                }
                 this.$store.dispatch(action, doc)
                     .then(function() {
                         self.$emit("submit");
                     });
             },
             onCopy: function(doc) {
                 if (doc._id) {
+                    if (doc.description === "-") {
+                        doc.description = "";
+                    }
+                    if (typeof (default_value) === 'object') {
+                        doc.default_value = JSON.stringify(doc.default_value);
+                    }
                     this.title = "Update parameter";
                     this.saveButtonLabel = "Save";
+                    if (doc.description) {
+                        this.showDescription = true;
+                    }
+                    if (doc.valuesList) {
+                        this.valuesList = doc.valuesList;
+                    }
+                    if (doc.expiry_dttm) {
+                        this.showExpirationDate = true;
+                    }
                 }
                 else {
                     this.title = "Add parameter";
                     this.saveButtonLabel = "Save";
+                    this.showDescription = false;
+                    this.valuesList = [];
+                    this.showExpirationDate = false;
                 }
+            },
+            openJsonEditor: function() {
+                this.$store.dispatch("countlyRemoteConfig/parameters/showJsonEditor", true);
             }
         }
     });
 
+    var ColorTag = countlyVue.views.BaseView.extend({
+        data: function() {
+            return {
+                selectedColorKey: this.defaultColorKey
+            };
+        },
+        methods: {
+            click: function(event) {
+                this.selectedColorKey = event.target.getAttribute('data-color');
```

Comment:
rather than accessing the data-color from event.target using the getAttribute,
you can pass the data-color value as argument to the click method.

Reply:
[this](https://github.com/Countly/countly-server/pull/2135/files?authenticity_token=vYyp91wk%2BqYqB42yV3zUT0B64NoqmNshpVSsQqPt%2BER9kAGFeXeaaYSrEqXsOEhYYzrO7L9dAKs5HjQ7WawFIQ%3D%3D&file-filters%5B%5D=.html&file-filters%5B%5D=.js&file-filters%5B%5D=.properties&file-filters%5B%5D=.scss&hide-deleted-files=true#diff-35a50fbf8b26bca06763311058cfc63614f4383b1e0e8b9bfa7995533959b53cR121) should be changed to 
<div v-if="tag.key == selectedColorKey" @click="click(tag)"

Reply:
Also move this component to helpers.js in core.

Reply:
Yes, We should try to use vue abstractions as much as possible and not interact with dom elements directly.

Reply:
Good catch. Yes that would be the ideal way :)